contracts/executable/AxelarGMPExecutableWithToken.sol

```diff
@@ -77,7 +117,7 @@ abstract contract AxelarGMPExecutableWithToken is IAxelarGMPExecutableWithToken,
      * @notice Returns the address of the IAxelarGMPGatewayWithToken contract.
      * @return The Axelar GMP Gateway with Token instance.
      */
-    function gatewayWithToken() internal view returns (IAxelarGMPGatewayWithToken) {
+    function gatewayWithToken() public view returns (IAxelarGMPGatewayWithToken) {
```

Comment:
i'm fine with keeping this internal. the app itself needs to use it, and not someone who calls the app. just `gateway()` can be external so a user can see the configured gateway on the contract

Reply:
The reason I changed it as `public` is I changed code in ITS : `gatewayWithToken()` is called in [InterchainTokenFactory](https://github.com/axelarnetwork/interchain-token-service/blob/7df851d8a4ec4df819961d704bf3067ee8d37521/contracts/InterchainTokenFactory.sol#L43) 

But yeah, we can keep it `internal` here and have an external function calling the `gatewayWithToken()` in `InterchainTokenService` for `InterchainTokenFactory`.

Reply:
yeah, should keep it internal. Also, we're removing the gateway token registration from the factory and adding it to the ITS hub side. So it won't be needed anymore